### PR TITLE
MULE-12219 Nested MessageProcessors post notifications are triggered within the error message

### PR DESCRIPTION
**Pull request by dfeist**
PENDING BUILD

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/internal/util/rx/IgnoreSubscriber.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * Singleton {@link Subscriber} that request unlimited demand and ignores errors.
+ * 
+ * @param <T>
+ */
+public class IgnoreSubscriber<T> extends BaseSubscriber<T> {
+
+  private static Subscriber INSTANCE = new IgnoreSubscriber<>();
+
+  public static <T> Subscriber<T> get() {
+    return new IgnoreSubscriber<>();
```
Comment (alepulver) [inline core/src/main/java/org/mule/runtime/core/internal/util/rx/IgnoreSubscriber.java]: Shouldn't this return INSTANCE?

File: core/src/main/java/org/mule/runtime/core/processor/chain/AbstractMessageProcessorChain.java [hunk 2]
```diff
@@ -103,16 +109,55 @@ protected Event doProcess(Event event) throws MuleException {
...
+            .onProcessor(processorFunction);
+      }
+
+      ReactiveProcessor finalProcessorFunction = processorFunction;
+      stream = from(stream).flatMap(event -> {
+        try {
+          return Flux.just(event)
+              .transform(finalProcessorFunction)
+              .onErrorResumeWith(MessagingException.class, handleError(event.getContext()));
+        } catch (Throwable throwable) {
```
Comment (alepulver) [inline core/src/main/java/org/mule/runtime/core/processor/chain/AbstractMessageProcessorChain.java]: Is this catch ever executed? If the flux is being used like a "builder" here, and not run (with force/subscribe/etc), exceptions thrown in finalProcessorFunction or handleError won't be caught here.

File: core/src/main/java/org/mule/runtime/core/internal/util/rx/IgnoreSubscriber.java (hunk 1 above)
Comment (dfeist) [inline core/src/main/java/org/mule/runtime/core/internal/util/rx/IgnoreSubscriber.java]: Oops yes.

File: core/src/main/java/org/mule/runtime/core/processor/chain/AbstractMessageProcessorChain.java (hunk 2 above)
Comment (dfeist) [inline core/src/main/java/org/mule/runtime/core/processor/chain/AbstractMessageProcessorChain.java]: This needs to be removed, forgot to remove it.

**Pull request merged**